tools/ReleaseNotes/release_editor.py

```diff
@@ -0,0 +1,129 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import os
+import sys
+import shutil
+
+DIRECTIVE = ".. amalgamate:: "
+
+
+def getReleaseRoot():
+    script_dir = os.path.split(sys.argv[0])[0]
+    return os.path.abspath(os.path.join(script_dir, '../../docs/source/release/'))
+
+
+def fixReleaseName(name):
+    if not name.startswith('v'):
+        name = 'v' + name
+    return name
+
+
+def createFileLocation(releaseNo):
+    release_root = getReleaseRoot()
+    release_root = release_root + '/' + releaseNo
+    return release_root
+
+
+def updateFile(originalFile, replaceText, textToReplace):
+    # opens the upper level release note e.g. diffraction.rst
+    mainRST = open(originalFile, 'r')
```

Comment:
Could I ask that we use the [with](https://realpython.com/python-with-statement/#the-with-statement-approach) statement for managing files, like is done below?

Reply:
I've changed all instances like this to usng the with statement.